examples/payment-channel/lib.rs

```diff
@@ -0,0 +1,557 @@
+
+    #[ink(storage)]
+    pub struct PaymentChannel {
+        /// The `AccountId` of the sender of the payment channel.
+        sender: AccountId,
+        /// The `AccountId` of the recipient of the payment channel.
+        recipient: AccountId,
+        /// The `Timestamp` at which the contract expires.
+        expiration: Timestamp,
+        /// The `Amount` withdrawn by the recipient.
+        withdrawn: Balance,
+        /// The `Timestamp` which will be added to the current time when the sender wishes to close the channel.
+        /// This will be set at the time of contract instantiation.
+        close_duration: Timestamp,
+    }
+
+    /// Errors that can occur upon calling this contract.
+    #[derive(Debug, PartialEq, Eq, scale::Encode, scale::Decode)]
+    #[cfg_attr(feature = "std", derive(::scale_info::TypeInfo))]
+    pub enum Error {
+        /// Returned if caller is not the `sender` while required to.
+        CallerIsNotSender,
+        /// Returned if caller is not the `recipient` while required to.
+        CallerIsNotRecipient,
+        /// Returned if the requested withdrawal / close amount is less than the amount that is already already withdrawn.
+        AmountIsLessThanWithdrawn,
+        /// Returned if the requested transfer failed. this can be the case if the contract does not have sufficient free
+        /// funds or if the transfer would have brought the contract's balance below minimum balance.
+        TransferFailed,
+        /// Returned if the contract hasn't expired yet and the `sender` wishes to close the channel.
+        NotYetExpired,
+    }
+
+    /// Type alias for the contract's result type.
+    pub type Result<T> = core::result::Result<T, Error>;
+
+    /// Emitted when the sender wishes to close the channel.
+    #[ink(event)]
+    pub struct SenderCloseStarted {
+        expiration: Timestamp,
+        close_duration: Timestamp,
+    }
+
+    impl PaymentChannel {
+        /// The only constructor of the contract.
+        ///
+        /// The recipient and the `close_duration` are required.
+        ///
+        /// `expiration` will be set to a max value so that the contract will never expire.
+        /// `sender` can call `start_sender_close` to override this.
+        /// `sender` will be able to claim the remaining balance by calling `claim_timeout` after `expiration` has passed.
+        #[ink(constructor)]
+        pub fn new(recipient: AccountId, close_duration: Timestamp) -> Self {
+            Self {
+                sender: Self::env().caller(),
+                recipient,
+                expiration: u64::pow(2, 63) - 1,
+                withdrawn: 0,
+                close_duration,
+            }
+        }
+
+        /// `recipient` can close the payment channel anytime. The `recipient` will be sent that amount,
+        /// and the remainder will go back to the `sender`.
+        #[ink(message)]
+        pub fn close(&mut self, amount: Balance, signature: [u8; 65]) {
```

Comment:
```suggestion
        pub fn close(&mut self, amount: Balance, signature: [u8; 65]) -> Result<()> {
```

Then as for your question, since we have a `Result<T>` we can call `unwrap()` to get the inner `T`. That should make the tests compile